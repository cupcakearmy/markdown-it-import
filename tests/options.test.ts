import { describe, expect, test } from 'bun:test'
import { render } from './utils.ts'

describe('options', () => {
  test('empty options', async () => {
    expect(await render('simple.md', {})).toMatchSnapshot()
  })

  test('set root directory', async () => {
    expect(await render('relative.md', { root: 'tests/fixtures' }))
  })

  describe('matcher', () => {
    test('custom matcher', async () => {
      const matcher = /foo '(?<file>.+)'/g
      expect(await render('matcher-a.md', { matcher })).toMatchSnapshot()
    })
    test('custom matcher', async () => {
      const matcher = /\$(?<file>.+)\$/g
      expect(await render('matcher-b.md', { matcher })).toMatchSnapshot()
    })

    test('fail for non global regexp', async () => {
      const matcher = /foo '(?<file>.+)'/
      expect(() => render('matcher-a.md', { matcher })).toThrow('RegExp must be global')
    })

    test('fail for missing "file" group', async () => {
      const matcher = /foo '(.+)'/g
      expect(() => render('matcher-a.md', { matcher })).toThrow('Regexp must expose a named group "file"')
    })
  })
})
