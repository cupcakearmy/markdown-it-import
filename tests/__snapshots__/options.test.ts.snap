// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`options matcher custom matcher 1`] = `
"<p>This is some amazing quote</p>
"
`;

exports[`options matcher custom matcher 2`] = `
"<p>This is some amazing quote</p>
"
`;

exports[`options empty options 1`] = `
"<h1>Some title</h1>
<ul>
<li>Some</li>
<li>List</li>
<li>Three</li>
</ul>
<blockquote>
<p>Some note</p>
</blockquote>
<table>
<thead>
<tr>
<th>Column A</th>
<th>Column B</th>
<th>Column C</th>
</tr>
</thead>
<tbody>
<tr>
<td>Some</td>
<td><code>stuff</code></td>
<td></td>
</tr>
</tbody>
</table>
"
`;
